### DISCUSSION TITLE
Class name must be a valid object or a string

### DISCUSSION THREAD

**Discussion by Traqza** [livewire/livewire#7020] [7. Bugs (archived)]
### Livewire version

3.0.6

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

Component

```
class ShowAllFileManagerTags extends Component
{
    public Collection $tags;

    public function mount(): void
    {
        $this->tags = (new FileTagManager)->orderBy('tag','ASC')->get();
    }

    public function render(): View|Application|Factory|\Illuminate\Contracts\Foundation\Application
    {
        return view('livewire.FileManager.show-all-file-manager-tags');
    }
}
```

View

```
<div>
    @foreach($tags as $tag)
        <a class="d-inline-block"
            style="display: inline; width: fit-content; padding: 2px;"
            href="{{ route('user.tagmanager.show', $tag->id) }}">
            <span style="background-color:{{ '#' . $tag->color_hex }}; font-weight:200;"
                class="badge text-dark fs-6">
                {{ $tag->tag }}
            </span>
        </a>
    @endforeach
</div>
```

Migration

```
 Schema::create('tags', function (Blueprint $table) {
            $table->uuid('id')->primary();
            $table->string('tag');
            $table->string('color_hex')->default('e4eaec');
            $table->timestamps();
            $table->softDeletes();
        });
```

Model
```
class FileTagManager extends Model
{
    use HasFactory;
    use SoftDeletes;

    protected $dates = ['deleted_at'];

    protected $table = 'tags';

    public $incrementing = false;

}
```

If the model has no objects in the database you will get an error

```Class name must be a valid object or a string```

Coming from the `$modelAlias = (new $modelClass)->getMorphClass();` on `.../Features/SupportModels/EloquentCollectionSynth.php`

If I add a tag to the database, page works as intended. If I remove tag, page crashes.

Confirmed to 3.0.6 by downgrading to 3.0.5 without errors with either empty or seeded table.

Let me know if you need more context or information. I am avaialble on Discord as wel.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by Traqza** [comment]
See https://github.com/livewire/livewire/pull/7017

**Comment by joshhanley** [comment]
@Traqza this has been fixed and released in v3.0.7 🙂
